software/metaphlan3/run/meta.yml

```diff
@@ -0,0 +1,57 @@
+name: metaphlan3_run
+description: MetaPhlAn is a tool for profiling the composition of microbial communities from metagenomic shotgun sequencing data.
+keywords:
+  - metagenomics
+  - classification
+  - fastq
+  - bam
+  - fasta
+tools:
+  - metaphlan3:
+      description:  Identify clades (phyla to species) present in the metagenome obtained from a microbiome sample and their relative abundance
+      homepage: None
+      documentation: None
+      tool_dev_url: None
+      doi: ""
+      licence: ['MIT License']
+
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - reads:
+      type: file
+      description: List of input FastQ files of size 1 and 2 for single-end and paired-end data, respectively.
+  - sam:
+      type: file
+      description: Bowtie2-produced SAM files containing read alignments to the MetaPHlAn marker database
+  - fasta:
+      type: file
+      description: Input files in FASTA format
+  - bowtie2out:
+      type: file
+      description: Intermediate Bowtie2 output produced from mapping the metagenome against the MetaPHlAn marker database ( not compatible with `bowtie2out` files generated with MetaPhlAn versions below 3 )
```

Comment:
I think it would be better to list all the input options under on entry called `input`, as that's how you called it in the `main.nf` script - this way it is clearer what is meant. Then you could add a pattern to indicated the different input options and just mention that different file inputs are possible in the description.

Reply:
Sounds good, have updated